apps/sms/views/conversation/js/conversation.js

```diff
@@ -1898,11 +1898,24 @@ var ConversationView = {
     }
     params.items.push(subjectItem);
 
-    // If we are on a thread, we can call to SelectMessages
+    /*
+     * If we are on a thread,
+     * we can call Delete Thread, SelectMessages or Message Settings
+     */
     if (Navigation.isCurrentPanel('thread')) {
       params.items.push({
+        l10nId: 'delete-thread',
+        method: () => {
+          InboxView.delete([this.activeThread.id]).then(() => {
```

Comment:
__issue:__ Please, don't introduce direct dependency to another view, we're going to split all these views in a separate documents and such dependencies make this harder to do. See https://github.com/mozilla-b2g/gaia/blob/master/apps/sms/views/conversation/js/startup.js#L54 where we shim such dependencies for the split views.

You likely need to generalize ```ConversationView.delete``` method to do the job.

Also delete confirmation text is not appropriate here - it will be ```Delete selected message thread?``` but we haven't selected anything :)